### PR TITLE
Avoid error messages/mod compat on duplicate entities

### PR DESCRIPTION
**Pull request by someaddons**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- Avoid error messages/mod compat on duplicate entities

[x ] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

**Review by Raycoms — APPROVED** [review]

File: src/main/java/com/minecolonies/core/event/EventHandler.java [hunk 1]
```diff
@@ -123,9 +124,15 @@ public static void onEntityAdded(@NotNull final EntityJoinLevelEvent event)
...
             {
-                ((Mob) event.getEntity()).targetSelector.addGoal(6, new NearestAttackableTargetGoal<>((Mob) event.getEntity(), EntityCitizen.class, true, citizen -> !citizen.isInvisible()));
+                ((Mob) event.getEntity()).targetSelector.addGoal(6,
+                  new NearestAttackableTargetGoal<>((Mob) event.getEntity(), EntityCitizen.class, true, citizen -> !citizen.isInvisible()));
                 ((Mob) event.getEntity()).targetSelector.addGoal(7, new NearestAttackableTargetGoal<>((Mob) event.getEntity(), EntityMercenary.class, true));
             }
+
+            if (event.getEntity() instanceof AbstractFastMinecoloniesEntity && ((ServerLevel) event.getLevel()).getEntity(event.getEntity().getUUID()) != null)
+            {
+                event.setCanceled(true);
```
Comment (uecasm) [inline src/main/java/com/minecolonies/core/event/EventHandler.java:134]: Cancelling the event entirely seems potentially troublesome.  It's still actually adding the entity to the world, and some things might want to be aware of that.  e.g. perhaps mapping mods or entity radars.

**Comment by uecasm** [comment]
Looking at the code of DEUF, the thing it seems to get upset about is when `level.getEntity(uuid)` doesn't match the entity that's being added.  Presumably, the real fix for this issue would be to delete the old entity strictly before spawning the new one?  Then there shouldn't be a conflict/duplication, and no need to cancel this event.

File: src/main/java/com/minecolonies/core/event/EventHandler.java (hunk 1 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/event/EventHandler.java:134]: image

**Comment by someaddons** [comment]
those are entities being added during chunk load, they cannot be the same object references as existing ones the colony respawned on tick

**Comment by uecasm** [comment]
Oh, so this is blocking the old persisted entities from respawning rather than affecting the intended respawn on button click/etc?  That makes more sense.

**Comment by someaddons** [comment]
yep we just manually drop the duplicate entity on load like vanilla does (duplicate uuid message) but by doing it manually we circumvent that "uuid fix" mod and the log message of vanilla

**Pull request merged**